airflow/models/dag.py

```diff
@@ -338,7 +338,7 @@ def _create_orm_dagrun(
     run.dag = dag
     # create the associated task instances
     # state is None at the moment of creation
-    run.verify_integrity(session=session)
+    run.verify_integrity(session=session, run_type=run_type)
```

Comment:
Instead of passing in run_type, this should just be a flag to skip `task_filter`.

Reply:
changed, can't completely skip task_filter otherwise it doesn't remove tasks properly but even with my current solution it still fails one test case

Reply:
it looks like the test depends on task.start_date <= self.execution_date, which gets overridden